src/viam/components/movement_sensor/movement_sensor.py

```diff
@@ -108,24 +110,41 @@ async def get_readings(self, *, extra: Optional[Dict[str, Any]] = None, timeout:
                 altitude: float,
                 linear_velocity: Vector3,
                 angular_velocity: Vector3,
+                linear_acceleration: Vector3
                 compass: float,
                 orientation: Orientation
             }
         """
-        ((pos, alt), lv, av, la, comp, orient) = await asyncio.gather(
+        (pos, lv, av, la, comp, orient) = await asyncio.gather(
             self.get_position(extra=extra, timeout=timeout),
             self.get_linear_velocity(extra=extra, timeout=timeout),
             self.get_angular_velocity(extra=extra, timeout=timeout),
             self.get_linear_acceleration(extra=extra, timeout=timeout),
             self.get_compass_heading(extra=extra, timeout=timeout),
             self.get_orientation(extra=extra, timeout=timeout),
+            return_exceptions=True,
         )
-        return {
-            "position": pos,
-            "altitude": alt,
-            "linear_velocity": lv,
-            "angular_velocity": av,
-            "linear_acceleration": la,
-            "compass": comp,
-            "orientation": orient,
-        }
+
+        readings = {}
+
+        # Add returned value to the readings dictionary if value is of expected type; omit if unimplemented.
+        def add_reading(name: str, reading, returntype: List) -> None:
+            if type(reading) in returntype:
+                if name == "position":
+                    readings["position"] = reading[0]
+                    readings["altitude"] = reading[1]
+                else:
+                    readings[name] = reading
+                return
+            elif isinstance(reading, GRPCError) and "Unimplemented" in str(reading.message):
```

Comment:
Add the local `NotImplementedError` `viam.errors.MethodNotImplementedError` and `Viam.errors.NotSupportedError`